### PR TITLE
fix: use deterministic LF line endings in generated files

### PR DESCRIPTION
**Pull request by joaodinissf**
Headless generation can emit platform-dependent line endings, producing CRLF on Windows despite the repository’s LF policy. This change makes the covered DDK generation paths emit LF consistently.

- Bind `LfLineSeparatorInformation` in the Check, CheckCfg, Export, Expression, Format and Scope runtime modules so Xtext’s file post-processors use LF.
- Set explicit LF delimiters in the ModelInference, TypeModel, test-language and HelloWorld generation workflows.
- Use an LF-terminated writer for keyword reports and diagnostics.
- Remove `LfNormalizingFileSystemAccess`: it normalized content before final post-processing and could discard trace-region information when converting generated content to a string.
- Update affected bundle and feature versions, including `com.avaloq.tools.ddk.xtext` to 17.4.0 for the new public API.

For downstream consumers, headless generation through these runtime bindings now emits LF on Windows. IDE writes with a resource URI continue to use the existing preference-based whitespace provider; its no-resource fallback uses the LF binding.

**Validation**

Tests cover the real headless file-write/post-processing path with LF and CRLF control bindings, Check and Format runtime injector wiring, CR-free Check-generated files, and report-writer terminators. The other four runtime bindings are not directly tested through their injectors.

The final source tree passed a full clean build on macOS with Java 21: **360 tests, zero failures or errors, two skipped**, plus Checkstyle, PMD/CPD, SpotBugs, baseline-version and line-ending checks.

Both intermediate commits also passed their full test and quality checks: **355 tests each, zero failures or errors, two skipped**. After the API-version and Javadoc follow-up, the final source tree was validated again with the full clean build and checks reported above.

### CODE REVIEW

File: com.avaloq.tools.ddk.xtext.valid/src/com/avaloq/tools/ddk/xtext/valid/ValidRuntimeModule.java [hunk 1]
```diff
@@ -24,6 +26,16 @@
  */
 public class ValidRuntimeModule extends com.avaloq.tools.ddk.xtext.valid.AbstractValidRuntimeModule {
 
+  /**
+   * Binds the generated-file line separator to LF so code generation is deterministic
+   * across platforms; headless builds otherwise fall back to the platform separator.
+   *
+   * @return the LF {@link ILineSeparatorInformation} implementation, never {@code null}
+   */
+  public Class<? extends ILineSeparatorInformation> bindILineSeparatorInformation() {
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.xtext.valid/src/com/avaloq/tools/ddk/xtext/valid/ValidRuntimeModule.java]: This language should have been removed, let us merge https://github.com/dsldevkit/dsl-devkit/pull/1523 first, and then update this PR

**Comment by rubenporras** [comment]
rebase?

**Review by rubenporras — APPROVED** [review]
